src/analysis/AnalysisClient.js

```diff
@@ -55,21 +61,27 @@ class AnalysisClient {
     although I have seen deals for >=3% consistently.
   */
   _getGCGROIData(pairedProduct, amazonFBACost){
-    let totalPaidToWalmartWithGCG = +(this.effectiveValueOfDollar * (+this._getTotalAmountPaidToWalmart(pairedProduct.walmartProd).toFixed(2))).toFixed(2);
-    let totalCostPerItemWithGCG = (totalPaidToWalmartWithGCG + amazonFBACost);
-    let dollarROIPerItemWithGCG = +(parseFloat(pairedProduct.amazonProd.price) - totalCostPerItemWithGCG).toFixed(2);
+    /*
+      to determine when to do comparison with amazonPrice or lowestOfferPrice
+      when doing Simple Cost analysis, the lowestOffer obj will be empty, so it will use amazon price
+      when doing secondary analysis, items with lowest offer will use lowestOffer price, other items will use amazonPrice
+    */
+    const amazonPrice = (Object.keys(pairedProduct.amazonProd.lowestOfferInfo).length === 0) ? pairedProduct.amazonProd.price : pairedProduct.amazonProd.lowestOfferInfo.lowestOfferInfo.Price.LandedPrice.Amount ;
```

Comment:
I saw you doing this above, but now that I see it again I'm going to comment on it. This is a very odd way to check if a lowest price offer exists. It's difficult to read and is not very robust. The `lowestOfferInfo` object should not be defined if no lowest offer exists. This means the AmazonProduct class should initialize it to null and then only define it once an offer is found.
That would change the conditional to simply be `pairedProduct.amazonProd.lowestOfferInfo` since null evaluates to false.